### PR TITLE
Services adjust

### PR DESCRIPTION
**Pull request by NiazBikbaev**

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/entity/Account.java [hunk 1]
```diff
@@ -55,4 +55,16 @@
     @JoinColumn(name = "account_id")
     private List<Subscription> subscriptions;
 
+    public void addSubscription(Subscription subscription) {
+        this.subscriptions.add(subscription);
+        if (subscription.getAccount() != this) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/entity/Account.java]: !subscription.getAccount().equals(this)

File: src/main/java/com/epam/magazinestore/entity/Subscription.java [hunk 2]
```diff
@@ -48,13 +51,15 @@
 
     @OneToOne(fetch = FetchType.EAGER, cascade = {CascadeType.DETACH, CascadeType.PERSIST, CascadeType.MERGE})
     @JoinColumn(name = "magazine_id")
+    @OnDelete(action = OnDeleteAction.CASCADE)
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/entity/Subscription.java]: Maybe you can add CascadeType.DELETE to OneToOne ?

File: src/test/java/com/epam/magazinestore/MagazineServiceTests.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        magazineService.deleteMagazine(MAGAZINE_ID);
+        List afterRemoveList = magazineService.getAvailableMagazines(PAGE_INDEX, PAGE_SIZE);
+        int newSize = afterRemoveList.size();
+        Assert.assertEquals(prevSize - 1, newSize);
+    }
+
+    @Test
+    @Rollback
+    public void getUnSubscribedMagazinesTest() {
+        Account account = accountService.getAccountByUsername("nyaz");
```
Comment (AntonKiselyov) [inline src/test/java/com/epam/magazinestore/MagazineServiceTests.java]: Test constant

File: src/main/java/com/epam/magazinestore/entity/Subscription.java (hunk 2 above)
Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/entity/Subscription.java]: Why? If we remove subscription,  magazine should not be deleted

Comment (NiazBikbaev) [reply src/main/java/com/epam/magazinestore/entity/Subscription.java]: ```@OnDelete``` adds on delete constraint to magazine, if we remove magazine, subscription will be removed cascadly

**Pull request merged**